src/snap_management.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Learn more at: https://juju.is/docs/sdk
+
+"""Snap Installation Module.
+
+Modified from https://github.com/canonical/k8s-operator/blob/main/charms/worker/k8s/src/snap.py
+"""
+
+
+import logging
+import platform
+
+import charms.operator_libs_linux.v2.snap as snap_lib
+from charms.operator_libs_linux.v2.snap import SnapState
+
+# Log messages can be retrieved using juju debug-log
+log = logging.getLogger(__name__)
+
+
+# Map of the grafana-agent snap revision to install for given architectures and strict mode.
+_grafana_agent_snap_spec = {
+    "strict": {
+        "amd64": {
+            "name": "grafana-agent",
+            "revision": "16",  # 0.35.0
+        },
+        "arm64": {
+            "name": "grafana-agent",
+            "revision": "23",  # 0.39.2
+        },
+    },
+}
+
+
+def install_ga_snap(classic: bool = False):
+    """Looks up system details and installs the appropriate grafana-agent snap revision."""
+    arch = get_system_arch()
+    confinement = "classic" if classic else "strict"
+    snap_spec = _grafana_agent_snap_spec[confinement][arch]
```

Comment:
why isn't there a `"classic"` section in the spec dict then? Is it fine if we raise a KeyError here?

Reply:
I think we need better exception management too, if the arch is unsupported. Better to catch and reraise something custom in that case.

Reply:
The `"classic"` section is not in the spec yet because that version of the charm is stuck in [ci purgatory](https://github.com/canonical/grafana-agent-snap/pull/66).  When that merges, we'll add a classic one as well

Reply:
added some validation for arch/classic so they raise a clear error, plus some tests